gamecore/src/main/java/com/mankomania/game/gamecore/client/MessageHandler.java

```diff
@@ -10,89 +10,94 @@
-
-        this.gameData.setPlayerToNewField(message.getPlayerId(), message.getFieldToMoveTo());
+        gameData.setPlayerToField(message.getPlayerIndex(), message.getFieldToMoveTo());
     }
 
     /**
      * Helper for sending the dice result to the server.
+     *
      * @param diceResult the rolled dice value
      */
     public void sendDiceResultMessage(int diceResult) {
-        Log.info("[sendDiceResultMessage] Got dice roll value from DiceScreen (" + diceResult + ").");
-        Log.info("[sendDiceResultMessage] Sending to server that local player (id: " + this.gameData.getLocalPlayer().getOwnConnectionId() + ") rolled a " + diceResult + ".");
+        Log.info("sendDiceResultMessage", "Got dice roll value from DiceScreen (" + diceResult + ").");
+        Log.info("sendDiceResultMessage", "Sending to server that local player (id: " + MankomaniaGame.getMankomaniaGame().getRefPlayer().getConnectionId() + ") rolled a " + diceResult + ".");
 
-        DiceResultMessage diceResultMessage = DiceResultMessage.createDiceResultMessage(this.gameData.getLocalPlayer().getOwnConnectionId(), diceResult);
-        this.client.sendTCP(diceResultMessage);
+        DiceResultMessage diceResultMessage = new DiceResultMessage(MankomaniaGame.getMankomaniaGame().getRefPlayer().getPlayerIndex(), diceResult);
+        client.sendTCP(diceResultMessage);
     }
 
     public void gotMoveToIntersectionMessage(MovePlayerToIntersectionMessage message) {
-        Log.info("[MovePlayerToIntersectionMessage] moving player to (" + message.getFieldToMoveTo() + ")");
-        Player playerMoving = this.gameData.getPlayerByConnectionId(message.getPlayerId());
-        playerMoving.setCurrentField(message.getFieldToMoveTo());
+        Log.info("gotMovePlayerToIntersectionMessage", "moving player to (" + message.getFieldIndex() + ")");
+
+        gameData.setPlayerToField(message.getPlayerIndex(), message.getFieldIndex());
 
-        this.gameData.setPlayerToNewField(message.getPlayerId(), message.getFieldToMoveTo());
+        Log.info("gotMovePlayerToIntersectionMessage", "need to send a path decision between (" + message.getSelectionOption1() + ") and (" + message.getSelectionOption2() + ")");
+        gameData.setIntersectionSelectionOption1(message.getSelectionOption1());
+        gameData.setIntersectionSelectionOption2(message.getSelectionOption2());
 
-        Log.info("[MovePlayerToIntersectionMessage] need to send a path decision between (" + message.getSelectionOption1() + ") and (" + message.getSelectionOption2() + ")");
-        this.gameData.setIntersectionSelectionOption1(message.getSelectionOption1());
-        this.gameData.setIntersectionSelectionOption2(message.getSelectionOption2());
-        // TODO: display UI to select intersection path
-        if(message.getPlayerId() == this.gameData.getLocalPlayer().getOwnConnectionId()){
+        if (message.getPlayerIndex() == MankomaniaGame.getMankomaniaGame().getRefPlayer().getPlayerIndex()) {

```

Comment:
rename `getRefPlayer` to something like `localPlayer(Ref)` so it's more clear that this player reference always represents the current local player on client side.